File: backend/app/models/project.py
from datetime import UTC, datetime

from bson import ObjectId
from pydantic import BaseModel, Field

from .py_object_id import PyObjectId


class Project(BaseModel):
    id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
    name: str
    description: str | None = None
    owner_id: PyObjectId
    members_ids: list[PyObjectId] = []
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
    meeting_datetime: datetime

    class Config:
        populate_by_name = True
        json_encoders = {ObjectId: str, PyObjectId: str}
        arbitrary_types_allowed = True

File: frontend/src/types/project.ts
export interface Project {
  _id: string;
  name: string;
  description: string;
  owner_id: string;
  members_ids: string[];
  created_at: string;
  updated_at: string;
}

export interface ProjectMember {
  _id: string;
  username: string;
}

export interface ProjectResponse {
  _id: string;
  name: string;
  description: string;
  owner: ProjectMember;
  members: ProjectMember[];
  created_at: string;
  updated_at: string;
}

export type ProjectUpdate = Partial<Project>;
